test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -4406,8 +4414,14 @@ await SimpleTestPackageUtility.CreateFolderFeedV3Async(
             result.LockFile.Targets[0].Libraries[1].Dependencies.Should().HaveCount(0);
             result.LockFile.Targets[0].Libraries[2].Name.Should().Be("C");
             result.LockFile.Targets[0].Libraries[2].Dependencies.Should().HaveCount(0);
-            result.LockFile.LogMessages.Should().HaveCount(1);
-            result.LockFile.LogMessages[0].Code.Should().Be(NuGetLogCode.NU1510);
+            if (shouldWarn)
+            {
+                result.LockFile.LogMessages.Should().HaveCount(1);
+                result.LockFile.LogMessages[0].Code.Should().Be(NuGetLogCode.NU1510);
+                result.LockFile.LogMessages[0].LibraryId.Should().Be("C");
+                result.LockFile.LogMessages[0].TargetGraphs.Should().HaveCount(1);
+                result.LockFile.LogMessages[0].TargetGraphs[0].Should().Be("net10.0");
```

Comment:
```suggestion
                result.LockFile.LogMessages[0].TargetGraphs[0].Should().Be(framework);
```

Should this match the param passed in?

Reply:
yeah, it could, that's better.